src/cdk/stepper/stepper.ts

```diff
@@ -210,6 +210,8 @@ export class CdkStepper {
     if (this._linear) {
       for (let i = 0; i < index; i++) {
         if (!stepsArray[i].stepControl.valid) {
+          // remove the focus from clicked step header if the step is not able to be selected
+          this._stepHeader.toArray()[index].nativeElement.blur();
```

Comment:
Nit: Hmm I can see why you did it this way, but blurring is a side effect I wouldn't expect from a function that simply reduces the validity.  I could see someone using it elsewhere without realizing it does this. Perhaps you can return the invalid step instead of true and blur in selectedIndex?